Dockerfile

```diff
@@ -0,0 +1,32 @@
+FROM registry.access.redhat.com/rhscl/ruby-23-rhel7
+
+# sshd
+USER root
+RUN yum install -y --setopt=tsflags=nodocs openssh-server libicu-devel && \
+    yum clean all && /usr/sbin/sshd-keygen
+
+RUN mkdir /var/run/sshd
+
+# TODO: is this needed? https://git.io/vrplM
+RUN sed 's@session\s*required\s*pam_loginuid.so@session optional pam_loginuid.so@g' -i /etc/pam.d/sshd
+
+# gems
+RUN ["bash", "-c", "gem install --no-ri --no-rdoc bunny"]
+
+# git user
+RUN groupadd -r git &&\
+     useradd -r -s /bin/bash -g git git && \
+     mkdir --parent /home/git && \
+     chown -R git:git /home/git
+
+# gitlab-shell setup
+COPY . /home/git/gitlab-shell
+WORKDIR /home/git/gitlab-shell
+RUN ["bash", "-c", "./bin/install"]
+
+COPY authorized_keys /home/git/.ssh/authorized_keys
+
+USER root
```

Comment:
The RUN instruction runs at build time, generally it doesn't make sense to run things like `systemctl start` because it won't persist any useful side effect in the built image.

If this image will require running as root, this will require customization in the OpenShift configuration to allow running containers as root... This means that every user of RHMAP would have to change their setup to allow running containers in a less secure mode.

We use nss_wrapper in some images, but I don't know if it helps with sshd.